### DISCUSSION TITLE
create_roles and sqlite.

### DISCUSSION THREAD

**Discussion by lgammo** [pallets-eco/flask-security#708] [Q&A]
I've been trying to create a role with this line:
       security.datastore.find_or_create_role(
            name="admin",
            description="Site Adminitrator",
            permissions={"admin-read", "admin-write", "customer-read", "customer-write"},
        )

The table has name/description/permissions fields as varargs.

I get this error:

(sqlite3.InterfaceError) Error binding parameter 2 - probably unsupported type.
[SQL: INSERT INTO application_roles (name, description, permissions) VALUES (?, ?, ?)]
[parameters: ('admin', 'Site Adminitrator', ['customer-write', 'admin-read', 'customer-read', 'admin-write'])]

Looking at the create_roles function, the permissions field is processes and sent to SQLAlchemy as a list. So this explains why I see a list in the last line in the error above.

The question is this:  is SQLAlchemy having problems with sqlite specifically, or is this a bug in flask-security? In other words, why is create_roles sending a list of strings to SQLAlchemy?

**Comment by lgammo** [comment]
In addition, I got the CLI to work and I tried to add a role and that worked, but adding permissions resulted in the same type unsupported error.

**Reply by jwag956** [reply]
In 5.0 a change was made to push the support of list types (such as permissions) down to the ORM layer.
Please read: https://flask-security-too.readthedocs.io/en/stable/changelog.html#backward-compatibility-concerns

If you use the models/fsqla* code - that should handle this case - if not - the release notes show the proper column definition.

If you still have issues - please include your model definitions.

**Reply by lgammo** [reply]
Thanks for the quick reply. 

I followed the changes in the change log and that certainly fixed my issues.
